static/app/views/dashboards/widgetBuilder/hooks/useWidgetBuilderState.tsx

```diff
@@ -122,10 +125,24 @@ function useWidgetBuilderState(): {
           setDescription(action.payload);
           break;
         case BuilderStateAction.SET_DISPLAY_TYPE:
+          setDisplayType(action.payload);
           if (action.payload === DisplayType.BIG_NUMBER) {
             setSort([]);
           }
-          setDisplayType(action.payload);
+          const [aggregates, columns] = partition(fields, field => {
+            const fieldString = generateFieldAsString(field);
+            return isAggregateFieldOrEquation(fieldString);
+          });
+          if (action.payload === DisplayType.TABLE) {
+            setYAxis([]);
+            setFields([...columns, ...aggregates, ...(yAxis ?? [])]);
```

Comment:
same comment about changing state and then using the state value here. Unless this is meant to be interchanged?

Reply:
I think in this case it's fine because I'm not relying on any of the pending state changes for downstream stuff. `fields` nor `yAxis` changing shouldn't be an issue here but rather I'm changing them

Reply:
okk!